### PR TITLE
Create the event stream adherence report.

### PR DESCRIPTION
**Pull request by alxdarksage**

### CODE REVIEW

File: src/main/java/org/sagebionetworks/bridge/models/schedules2/adherence/eventstream/EventStreamAdherenceReport.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import java.util.List;
+
+import org.joda.time.DateTime;
+import org.sagebionetworks.bridge.json.DateTimeSerializer;
+
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+@JsonPropertyOrder({ "activeOnly", "timestamp", "clientTimeZone", "adherencePercentage", "dayRangeOfAllStreams",
+        "streams", "type" })
```
Comment (alxdarksage) [inline src/main/java/org/sagebionetworks/bridge/models/schedules2/adherence/eventstream/EventStreamAdherenceReport.java:13]: Check the order of these fields

File: src/main/java/org/sagebionetworks/bridge/models/schedules2/adherence/eventstream/EventStreamAdherenceReportGenerator.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        adherenceByEventId = builder.adherenceRecords.stream()
+                .collect(toMap(AdherenceRecord::getInstanceGuid, (a) -> a));
+        
+        LocalDate localNow = (builder.now != null) ? builder.now.toLocalDate() : null;
+        for (StudyActivityEvent event : builder.events) {
+            DateTime eventTimestamp = event.getTimestamp();
+            if (event.getClientTimeZone() != null) {
+                eventTimestamp = eventTimestamp.withZone(DateTimeZone.forID(event.getClientTimeZone()));
+            } else {
+                eventTimestamp = eventTimestamp.withZone(builder.now.getZone());
```
Comment (alxdarksage) [inline src/main/java/org/sagebionetworks/bridge/models/schedules2/adherence/eventstream/EventStreamAdherenceReportGenerator.java]: Should we directly use the clientTimeZone being passed in?

File: src/main/java/org/sagebionetworks/bridge/models/schedules2/adherence/eventstream/EventStreamAdherenceReport.java (hunk 1 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/models/schedules2/adherence/eventstream/EventStreamAdherenceReport.java:13]: changed to adherencePercent, field now where I'd expect it

File: src/main/java/org/sagebionetworks/bridge/models/schedules2/adherence/eventstream/EventStreamAdherenceReportGenerator.java (hunk 2 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/models/schedules2/adherence/eventstream/EventStreamAdherenceReportGenerator.java]: checks event zone, participant's persisted zone, and the zone of "now" in that order. Tests added for this.

File: src/test/java/org/sagebionetworks/bridge/models/schedules2/adherence/eventstream/EventStreamTest.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        assertEquals(deser.getDaysSinceEvent(), Integer.valueOf(5));
+        assertEquals(deser.getStudyBurstId(), "studyBurstId");
+        assertEquals(deser.getStudyBurstNum(), Integer.valueOf(2));
+        assertEquals(deser.getByDayEntries().size(), 2);
+        assertEquals(deser.getByDayEntries().get(6).size(), 2);
+        assertEquals(deser.getByDayEntries().get(8).size(), 1);
+    }
+    
+    @Test
+    public void retrieveEventStream() {
```
Comment (jnelsonjava) [inline src/test/java/org/sagebionetworks/bridge/models/schedules2/adherence/eventstream/EventStreamTest.java]: I can't tell what this is testing. Is it just verifying that the setters take the expected input?

**Review by jnelsonjava — APPROVED** [review]
Looks good. There's a comment on one test you might want to double check.

File: src/test/java/org/sagebionetworks/bridge/models/schedules2/adherence/eventstream/EventStreamTest.java (hunk 3 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/models/schedules2/adherence/eventstream/EventStreamTest.java]: It's testing nothing! Removed this.

**Pull request merged**